test/e2e/apimachinery/webhook.go

```diff
@@ -1727,21 +1699,19 @@ func registerWebhookForCustomResource(f *framework.Framework, configName string,
 	})
 	framework.ExpectNoError(err, "registering custom resource webhook config %s with namespace %s", configName, namespace)
 
-	err = waitWebhookConfigurationReady(f)
+	err = waitWebhookConfigurationReady(ctx, f)
 	framework.ExpectNoError(err, "waiting for webhook configuration to be ready")
-	return func() {
-		client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete(context.TODO(), configName, metav1.DeleteOptions{})
-	}
+	ginkgo.DeferCleanup(framework.IgnoreNotFound(client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete), configName, metav1.DeleteOptions{})
```

Comment:
these are the behavioural changes I'd prefer to not do at the same time, this wasn't asserting any error before ... maybe we can avoid all these `ginkgo.DeferCleanup` changes and do only the contexts in this commit?

Reply:
Deferring the usage of `ginkgo.DeferCleanup` means that all of these lines must be touched twice.

Reply:
Let me try one thing: I'll do a pass through the PR and extract those parts that introduce `ginkgo.DeferCleanup`. It should be possible to merge those parts first.

It will take time, though, because in several places I'll then conflict with my own follow-up changes.

Reply:
that will be great, thanks, just the context only are straightforward to review and approve ...